modyn/supervisor/internal/triggers/drift/decision_policy.py

```diff
@@ -41,27 +41,58 @@ def evaluate_decision(self, distance: float) -> bool:
 
 class DynamicDecisionPolicy(DriftDecisionPolicy):
     """Decision policy that will make the binary is_drift decisions based on a
-    dynamic threshold.
+    dynamic threshold."""
+
+    def __init__(self, config: DynamicThresholdCriterion):
+        self.config = config
+        self.score_observations: deque = deque(maxlen=self.config.window_size)
+
+
+class DynamicPercentileThresholdPolicy(DynamicDecisionPolicy):
```

Comment:
I also adjusted this decision policy to align with my (unpushed) implementation in my offline drift exploration. Seems more bulletproof and easier to understand.